### ISSUE TITLE
Display of activation key on bootstrap minion form

### ISSUE THREAD

**Issue by TimShaw61** [uyuni-project/uyuni#2122]
On the bootstrap minions form the selection box for the activation key only displays the key itself and not the associated description/label.  This is unhelpful as no-one can remember the which actual keys relate to the required software sources...

![bootstrapping](https://user-images.githubusercontent.com/17780891/79316239-f9240580-7efb-11ea-9ec6-10e1ae606376.png)

The description or label should be displayed as a choice not the actual key itself

**Comment by admd** [comment]
Hi @TimShaw61 ,
There is no field `label` and `description` is an optional field so we can't depend on it. Only field which can uniquely identify is key and that's is what we display here.

**Comment by TimShaw61** [comment]
Yes, but how is the user supposed to select a key? Are they supposed to just memorise each key and to which OS it refers?  It's ridiculous.  If I were programming this I would either display the key and the description or make the description field mandatory.  In a sense it already has a value because the note below the description input text box states:If left blank, this field will be filled in 'None'.So just make it mandatory for the user to enter a value and give us something sensible to choose from when selecting.  If the user is stupid enough not to add a description that's their fault - but as it stands I have to have a piece of paper with a lookup table so I can select the right one.  A bit silly...T
-------- Original message --------From: Abid Mehmood <notifications@github.com> Date: 08/05/2020  16:11  (GMT+00:00) To: uyuni-project/uyuni <uyuni@noreply.github.com> Cc: Tim Shaw <tim.shaw@medsci.ox.ac.uk>, Mention <mention@noreply.github.com> Subject: Re: [uyuni-project/uyuni] Display of activation key on bootstrap minion form (#2122) 

Hi 
@TimShaw61 ,
There is no field label and description is an optional field so we can't depend on it. Only field which can uniquely identify is key and that's is what we display here.
—
You are receiving this because you were mentioned.
Reply to this email directly, 
view it on GitHub, or 
unsubscribe.

**Comment by TimShaw61** [comment]
Yes, but how is the user supposed to select a key? Just saying that's the only field available isn't enough if the contents of that field make no sense as a selection criterion!

Are they supposed to just memorise each key and to which OS it refers?  It's ridiculous.  If I were programming this I would either display the key AND the description or just the description and make the description field mandatory.  It always has a value because the note below the description input text box states:


If left blank, this field will be filled in 'None'.


So just make it mandatory for the user to enter a value and give us something sensible to choose from when selecting.  If the user is stupid enough not to add a description that's their fault - but as it stands I have to have a piece of paper with a lookup table so I can select the right one which is a bit silly...


T

Sent from Outlook Mobile<https://aka.ms/blhgte>

________________________________

**Comment by juliogonzalez** [comment]
> Are they supposed to just memorise each key and to which OS it refers?  It's ridiculous.  If I were programming this I would either display the key AND the description or just the description and make the description field mandatory.  It always has a value because the note below the description input text box states:

While not ideal, the activation key list can be opened in parallel, as the description is there:

![image](https://user-images.githubusercontent.com/4226070/81427853-abe22f00-915b-11ea-8f4a-a093be4e7292.png)

Still, FMPOV the activation key name itself should be usually descriptive enough itself.

The problem I see here is that the description is just free text (I am not even sure what the limit is), so not sure how trivial would be showing this together with the key name. @ncounter opinions?

**Comment by TimShaw61** [comment]
I tend to stick with the activation key as offered when created which is
a long random string although I do make it shorter. I suppose I could
simply tack on something descriptive into the key so

1-23ab48de37fe.... could be changed to

1-CENTOS723ab48de37de... for example

Perhaps this is what everyone does but I've been too stupid to realise
it...

T




Sent from my BlackBerry 10 smartphone on the O2 network.
   From: Julio González Gil
Sent: Friday, 8 May 2020 17:46
To: Tim Shaw; uyuni-project/uyuni
Reply To: uyuni-project/uyuni
Cc: Mention
Subject: Re: [uyuni-project/uyuni] Display of activation key on
bootstrap minion form (#2122)




Are they supposed to just memorise each key and to which OS it refers?
It's ridiculous. If I were programming this I would either display the
key AND the description or just the description and make the description
field mandatory. It always has a value because the note below the
description input text box states:

While not ideal, the activation key list can be opened in parallel, as
the description is there:



Still, FMPOV the activation key name itself should be usually
descriptive enough itself.

The problem I see here is that the description is just free text (I am
not even sure what the limit is), so not sure how trivial would be
showing this together with the key name.@ncounter opinions?

**Comment by juliogonzalez** [comment]
> Perhaps this is what everyone does but I've been too stupid to realise it...

No, not really. You are using the product on a supported use-case.

But to be honest, since I started using it, I never use that random generation so I completely forgot about it.

I wonder if it makes sense at all having that random generation, and what's the benefit. I think having "human" names makes much more sense.

@ncounter any idea?
